WorkingFromHome/README.md

```diff
@@ -0,0 +1,54 @@
+
+This README.md contains only architecture and installation steps. For more details, please check the blog series [part 1](https://medium.com/i-love-my-local-farmer-engineering-blog/working-from-home-1-2-748d9acc3fb8), [part 2](https://medium.com/i-love-my-local-farmer-engineering-blog/working-from-home-2-2-53142394620b) 
+
+## Prerequisite
+1. **AWS CDK (v 1.102.0)**: This solutions use AWS CDK for deployment. If you aren't familiar with CDK, please follow this [Java CDK workshop](https://cdkworkshop.com/50-java.html) first. 
+2. **JDK 8 or later**: If you haven't used installed this version before, you can use [Amazon Corretto 8 distribution](https://aws.amazon.com/corretto/)
+3. **Maven**: For building the CDK code into assets for deployment. 
+
+## Architecture
+To understand what this solution will deploy, we will start from the original architecture.
+
+![](.README_images/on-prem-architecture.png)
+
+The on-prem Paris Data Center connects to Paris Office via a lease line, but has VPN connection for other office (as they have lower traffic). The VPN connector cannot scale beyond 50 IPSec clients, so we need to change its architecture to support additional VPN connections from employees.
+
+![](.README_images/on-aws.png)
+
+The solution uses AWS Client VPN to handle connections from employees' laptops. Then it forwards traffic to the Paris Data Center via AWS Site-to-Site VPN. AD Connector handles authentication and authorization.
+
+## Components 
+
+The main components of these solutions are in separate folders:
+1. `SiteToSiteVpnSetup` will set up AWS Site-to-Site VPN and an AWS VPC that we'll deploy AWS client VPN.  
+1. `ClientVpnSetup` will set up AWS Client VPN alongs with AD Connector.
+1. `CustomResourceUtils` contains reusable code for creating a custom resource. This is used by `ClientVpnSetup` for AD setup.
+
+## Installation steps
+1. `SiteToSiteVpnSetup` component
+    1. Modify `SiteToSiteVpnSetup`'s [cdk.context.json](WorkingFromHome/SiteToSiteConnection/cdk.context.json) with your on-prem IP address, CIDR range, and AWS region/availability zones.
+    1. Change the account an region in the file [SiteToSiteConnectionApp.java](WorkingFromHome/SiteToSiteConnection/src/main/java/com/ilmlf/sitetositeconnection/SiteToSiteConnectionApp.java)
+    1. Deploy with
+    ```
+    cd SiteToSiteVpnSetup
+    mvn package
+    cdk deploy 
+    ```
+1. Manually create a secret in Secret Manager with name `DomainAdminPassword` and put AD admin password in it. Create a [resource-based policy](https://docs.aws.amazon.com/secretsmanager/latest/userguide/auth-and-access_resource-based-policies.html) to limit the access on who could read the value.
```

Comment:
What's the structure of the secret?  I assume that it's plaintext here.

Reply:
Wondering if we could create this with CDK in the first stack with least privilege. It is a pain to do this manually, but too important to ignore.

Reply:
This is the Admin password of the on premise AD so it would reside on prem so can't really be created by any of the aws stacks

Reply:
In our blog we already say that we create the step manually. The reason being that it is easier rotated if created by itself in the console. If it was part of the CDK deployment it could be a pain to rotate it without possible affecting other parts of the stack

Reply:
Fair point. They have different life cycle. Resolving this.